opt_einsum/paths.py

```diff
@@ -88,6 +89,62 @@ def optimal(input_sets, output_set, idx_dict, memory_limit):
     return path
 
 
+def _calc_k12_flops(inputs, output, remaining, i, j, size_dict):
+    k1, k2 = inputs[i], inputs[j]
+    either = k1 | k2
+    two = k1 & k2
+    one = either - two
+    keep = frozenset.union(output, *(inputs[p] for p in remaining if p not in (i, j)))
+
+    k12 = one | (two & keep)
```

Comment:
Hmm can you help me understand this? I would think
```
k12 == either & keep == (one | two) & keep
```
whereas `one | (two & keep)` could accidentally preserve a dim that should not be kept 😕

Reply:
Yes I up ending confusing myself when writing this (then it worked for my contractions so I forgot about it!) - glad you had a look over. I was coming from the standard tensor notation where in general ``k12 = one``, then I added in ``two & keep`` to take care of triple+ indices, forgetting that ``output`` is already processed and in ``keep``. I'll change to:
```python
k12 = either & keep
```
Some kind of ``dim_ref_count`` as you had in ``'eager'`` might be nice here, the trick is just that each of  the factorially many paths would need its own - a bit cumbersome.